src/core/EventBus.js

```diff
@@ -46,14 +47,18 @@ function EventBus() {
             throw new Error('listener must be a function: ' + listener);
         }
 
+        let priority = options.priority || EVENT_PRIORITY_LOW;
+        let mode = options.mode || EVENT_MODE_ON_START;
```

Comment:
I suggest to not define a default mode as this is only used for DASH events. Instead omit the mode in the handler and check in the trigger function if mode is !== undefined and matches the target mode. Similar what is done with streamId and mediaType